### DISCUSSION TITLE
如何在一条消息发送后几秒内撤回消息

### DISCUSSION THREAD

**Discussion by Rhhhhhhh** [nonebot/discussions#219] [问答] [answered]
想在某些特定条件下发送一条消息后一段时间撤回，但是如果在一个event里面好像并不能获取刚发送的消息的message_id，请问是否只能通过另起一个函数在结束上述事件结束之后获取上一个事件中bot发送消息的message_id并对消息进行撤回？

**Comment by yanyongyu** [comment, marked as answer]
有几种方法：

1. 直接在发送后使用`asyncio.sleep`等待一段时间，然后使用send后返回的message id去撤回
2. 使用 apscheduler 插件新增一个定时任务，定时在一段时间后
3. 使用 `loop = asyncio.get_running_loop(); loop.call_later(123, func)` 创建一个任务

**Comment by wiketool** [comment]
Example code
```python
    res = await bind_matcher.send(reply)
    if res['message_id'] is not None:
        scheduler.add_job(bot.delete_msg, 'date',
                          run_date=datetime.now() + timedelta(seconds=5),
                          kwargs={'message_id': res['message_id']})
```
